File: android/src/main/java/com/lightningkite/kwift/captureWeak.shared.kt
package com.lightningkite.kwift

import com.lightningkite.kwift.AnyObject
import com.lightningkite.kwift.escaping
import com.lightningkite.kwift.weak


fun <Z : AnyObject> captureWeak(capture: Z, lambda: @escaping() (Z) -> Unit): () -> Unit {
    val captured by weak(capture)
    return label@{ ->
        val actualCaptured = captured
        if (actualCaptured == null) {
            return@label
        }
        lambda(actualCaptured!!)
    }
}

fun <Z : AnyObject, A> captureWeak(capture: Z, lambda: @escaping() (Z, A) -> Unit): (A) -> Unit {
    val captured by weak(capture)
    return label@{ a ->
        val actualCaptured = captured
        if (actualCaptured == null) {
            return@label
        }
        lambda(actualCaptured!!, a)
    }
}

fun <Z : AnyObject, A, B> captureWeak(capture: Z, lambda: @escaping() (Z, A, B) -> Unit): (A, B) -> Unit {
    val captured by weak(capture)
    return label@{ a, b ->
        val actualCaptured = captured
        if (actualCaptured == null) {
            return@label
        }
        lambda(actualCaptured!!, a, b)
    }
}

fun <Z : AnyObject, A, B, C> captureWeak(capture: Z, lambda: @escaping() (Z, A, B, C) -> Unit): (A, B, C) -> Unit {
    val captured by weak(capture)
    return label@{ a, b, c ->
        val actualCaptured = captured
        if (actualCaptured == null) {
            return@label
        }
        lambda(actualCaptured!!, a, b, c)
    }
}

fun <Z : AnyObject, A, B, C, D> captureWeak(
    capture: Z,
    lambda: @escaping() (Z, A, B, C, D) -> Unit
): (A, B, C, D) -> Unit {
    val captured by weak(capture)
    return label@{ a, b, c, d ->
        val actualCaptured = captured
        if (actualCaptured == null) {
            return@label
        }
        lambda(actualCaptured!!, a, b, c, d)
    }
}

fun <Z : AnyObject, A, B, C, D, E> captureWeak(
    capture: Z,
    lambda: @escaping() (Z, A, B, C, D, E) -> Unit
): (A, B, C, D, E) -> Unit {
    val captured by weak(capture)
    return label@{ a, b, c, d, e ->
        val actualCaptured = captured
        if (actualCaptured == null) {
            return@label
        }
        lambda(actualCaptured!!, a, b, c, d, e)
    }
}

File: ios/Kwift/com/lightningkite/kwift/captureWeak.shared.swift
//Package: com.lightningkite.kwift
//Converted using Kwift2

import Foundation



public func captureWeak<Z: AnyObject>(capture: Z, lambda: @escaping (Z) -> Void) -> () -> Void {
    weak var captured = capture
    return { () in 
        var actualCaptured = captured
        if actualCaptured == nil {
            return
        }
        lambda(actualCaptured!)
    }
}
public func captureWeak<Z: AnyObject>(_ capture: Z, _ lambda: @escaping (Z) -> Void) -> () -> Void {
    return captureWeak(capture: capture, lambda: lambda)
}
 
 

public func captureWeak<Z: AnyObject, A>(capture: Z, lambda: @escaping (Z, A) -> Void) -> (A) -> Void {
    weak var captured = capture
    return { (a) in 
        var actualCaptured = captured
        if actualCaptured == nil {
            return
        }
        lambda(actualCaptured!, a)
    }
}
public func captureWeak<Z: AnyObject, A>(_ capture: Z, _ lambda: @escaping (Z, A) -> Void) -> (A) -> Void {
    return captureWeak(capture: capture, lambda: lambda)
}
 
 

public func captureWeak<Z: AnyObject, A, B>(capture: Z, lambda: @escaping (Z, A, B) -> Void) -> (A, B) -> Void {
    weak var captured = capture
    return { (a, b) in 
        var actualCaptured = captured
        if actualCaptured == nil {
            return
        }
        lambda(actualCaptured!, a, b)
    }
}
public func captureWeak<Z: AnyObject, A, B>(_ capture: Z, _ lambda: @escaping (Z, A, B) -> Void) -> (A, B) -> Void {
    return captureWeak(capture: capture, lambda: lambda)
}
 
 

public func captureWeak<Z: AnyObject, A, B, C>(capture: Z, lambda: @escaping (Z, A, B, C) -> Void) -> (A, B, C) -> Void {
    weak var captured = capture
    return { (a, b, c) in 
        var actualCaptured = captured
        if actualCaptured == nil {
            return
        }
        lambda(actualCaptured!, a, b, c)
    }
}
public func captureWeak<Z: AnyObject, A, B, C>(_ capture: Z, _ lambda: @escaping (Z, A, B, C) -> Void) -> (A, B, C) -> Void {
    return captureWeak(capture: capture, lambda: lambda)
}
 
 

public func captureWeak<Z: AnyObject, A, B, C, D>(capture: Z, lambda: @escaping (Z, A, B, C, D) -> Void) -> (A, B, C, D) -> Void {
    weak var captured = capture
    return { (a, b, c, d) in 
        var actualCaptured = captured
        if actualCaptured == nil {
            return
        }
        lambda(actualCaptured!, a, b, c, d)
    }
}
public func captureWeak<Z: AnyObject, A, B, C, D>(_ capture: Z, _ lambda: @escaping (Z, A, B, C, D) -> Void) -> (A, B, C, D) -> Void {
    return captureWeak(capture: capture, lambda: lambda)
}
 
 

public func captureWeak<Z: AnyObject, A, B, C, D, E>(capture: Z, lambda: @escaping (Z, A, B, C, D, E) -> Void) -> (A, B, C, D, E) -> Void {
    weak var captured = capture
    return { (a, b, c, d, e) in 
        var actualCaptured = captured
        if actualCaptured == nil {
            return
        }
        lambda(actualCaptured!, a, b, c, d, e)
    }
}
public func captureWeak<Z: AnyObject, A, B, C, D, E>(_ capture: Z, _ lambda: @escaping (Z, A, B, C, D, E) -> Void) -> (A, B, C, D, E) -> Void {
    return captureWeak(capture: capture, lambda: lambda)
}
 
